src/dataflow/src/sink/kafka.rs

```diff
@@ -856,9 +1129,50 @@ where
                     accum.meet(&history.durability_frontier())
                 });
 
+            // Can't use `?` when the return type is a `bool` so use a custom try operator
+            macro_rules! bail_err {
+                ($expr:expr) => {
+                    match $expr {
+                        Ok(val) => val,
+                        Err(_) => {
+                            s.activator.activate();
```

Comment:
I think I missed this in the previous PR. Why do we have to re-activated immediately on error?

Reply:
What I'm trying to say is that if there logic expressed in `async_op` needs to describe some sort of "retry if an error happens logic" you could wrap everything in a `loop {}` block (or use one of the existing while loops) to break to (either implicitly or with a label).

The pattern here is as if we're implementing a `break 'start_of_function` using activations

Reply:
Returning from this function is required to move the frontier forward so I think (for now at least?) this approach is preferable -- what do you think?